src/dex-ui/store/swapSlice/swapSlice.ts

```diff
@@ -139,12 +140,21 @@ const createSwapSlice: SwapSlice = (set, get): SwapStore => {
         if (precision === undefined) {
           throw Error("Precision not found");
         }
-        const spotPriceL49BToL49A = await DexService.fetchSpotPrice(selectedAccountId);
-        const spotPriceL49AToL49B = spotPriceL49BToL49A ? BigNumber(1).dividedBy(spotPriceL49BToL49A) : undefined;
-        const spotPriceL49AToL49BWithPrecision = spotPriceL49AToL49B ? spotPriceL49AToL49B.times(precision) : undefined;
-        const spotPriceL49BToL49AWithPrecision = spotPriceL49BToL49A
-          ? spotPriceL49BToL49A.dividedBy(precision)
+        const [spotPriceTokenBToTokenA, pairTokenIds] = await Promise.all([
+          DexService.fetchSpotPrice(selectedAccountId),
```

Comment:
DexService.fetchSpotPrice always returns the same value for a pair contract regardless of the pair ordering in the UI. Swapping with the previous logic would mismatch the spot prices in the zustand store.

Example:

Values before switching token inputs:
B=>A = 1.1
A=>B = 0.9

Actual values after switching token inputs
B=>A = 0.9
A=>B = 1.1

Expected values after switching token inputs (spot price mappings stay the same). The new logic follows this approach.
B=>A = 1.1
A=>B = 0.9